### PR TITLE
Adds GitHub Teams integration

### PR DESCRIPTION
**Pull request by ttomsu**
Please review only the second commit.

### CODE REVIEW

File: actions/generate-schedule/main.go [hunk 1]
```diff
@@ -112,8 +131,18 @@ func validateFlags() error {
 		}
 	}
 
-	if *userList == "" {
-		return fmt.Errorf("--users must be specified")
+	if *githubTeam != "" || *githubOrg != "" {
```
Comment (plumpy) [inline actions/generate-schedule/main.go:170]: also check that userList is unset in here

File: actions/generate-schedule/main.go [hunk 2]
```diff
@@ -112,8 +131,18 @@ func validateFlags() error {
...
+			return fmt.Errorf("must specify both --githubTeam and --githubOrg")
+		}
+
+		var ok bool
+		accessToken, ok = os.LookupEnv(githubTokenEnvKey)
+		if !ok || accessToken == "" {
+			log.Fatalf("GITHUB_TOKEN environment variable cannot be empty. Get one from https://github.com/settings/tokens, and ensure it has the 'read:org' scope.")
+		}
+	} else if *userList == "" {
+		log.Fatalf("must specify either non-empty --users or (--githubTeam and --githubOrg).")
```
Comment (maggieneterval) [inline actions/generate-schedule/main.go:183]: Was about to comment that perhaps we could add this sentence to the validation for each flag just to make it clear the conditions under which each is required, but if we only anticipate a single client then not sure it's worth it 🤷‍♀

**Review by maggieneterval — APPROVED** [review]
:octocat: 👨‍👧‍👦 🤝

**Review by plumpy — APPROVED** [review]

File: actions/generate-schedule/main.go (hunk 1 above)
Comment (ttomsu) [reply actions/generate-schedule/main.go:170]: Done.

File: actions/generate-schedule/main.go (hunk 2 above)
Comment (ttomsu) [reply actions/generate-schedule/main.go:183]: Added. Also merged in the `--record` logic like I did in the Google Calendar PR

**Pull request merged**